pkg/apis/triggers/config/allowed_types.go

```diff
@@ -0,0 +1,30 @@
+package config
+
+import (
+	pipelinev1alpha1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	pipelinev1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+)
+
+var Decoder runtime.Decoder
+
+// TODO(dibyom): We should have a way of configuring this instead of an init function?
+func init() {
+	scheme := runtime.NewScheme()
+	utilruntime.Must(pipelinev1alpha1.AddToScheme(scheme))
+	utilruntime.Must(pipelinev1beta1.AddToScheme(scheme))
+	codec := serializer.NewCodecFactory(scheme)
+	Decoder = codec.UniversalDecoder(
+		pipelinev1alpha1.SchemeGroupVersion,
+		pipelinev1beta1.SchemeGroupVersion,
+	)
+}
+
+// IsAllowedType returns nil if the resourceTemplate has an apiVersion
```

Comment:
`IsAllowed` implies a `bool` response?

Reply:
I would also be in favour of e.g. `EnsureAllowedType` or something similar. `IsSomething` always implies a bool response to me as well.

Reply:
Switched to EnsureAllowedType. Thanks!